watcher.go

```diff
@@ -0,0 +1,120 @@
+package frankenphp
+
+import (
+	fswatch "github.com/dunglas/go-fswatch"
+	"go.uber.org/zap"
+	"sync/atomic"
+	"sync"
+)
+
+// latency of the watcher in seconds
+const watcherLatency = 0.15
+
+var (
+	watchSessions      []*fswatch.Session
+	// we block reloading until workers have stopped
+	blockReloading     atomic.Bool
+	// when stopping the watcher we need to wait for reloading to finish
+	reloadWaitGroup    sync.WaitGroup
+	// the integrity ensures rouge events are ignored
+	watchIntegrity	   atomic.Int32
+)
+
+func initWatcher(watchOpts []watchOpt, workerOpts []workerOpt) error {
+	if(len(watchOpts) == 0 || len(workerOpts) == 0) {
+		return nil
+	}
+
+	watchIntegrity.Store(watchIntegrity.Load() + 1)
```

Comment:
I'm not sure this works how you think it works. It basically allows only a single watcher to work at a time (the last one registered) and appears to cause some sneaky race conditions all over the place. Instead, you should use a slice.

Reply:
This actually solves a race condition since the C library sometimes still sends events even after it was shut down.
But if we want multiple watchers at a time, I can turn it into a slice (and call it watcher id or something)

Reply:
Could that be by design? Maybe the file changes were "queued up" before the shutdown? In any case, it will probably be a simpler implementation to just handle the events as they come in, even if the watcher is shut down.

Reply:
In other words, a simple boolean: `isEnabled` or something. Then if it is true, handle the event. On shutdown, set it to false.

Reply:
That was how I originally implemented it, but it caused tests to fail. Frankenphp would shut down completely and restart between tests, but the next test would receive file change events from a previous test. That essentially made the '**don't reload workers if file should not be watched**' test fail. Adding some kind of identifier like this was the only way I could think of to resolve that.
Outside of tests this probably doesn't matter since frankenphp is not restarted in such quick succession

Reply:
Found a better workaround for this 👍